### PR TITLE
feat(accordion): add meta

### PR DESCRIPTION
**Pull request by rachelbt**
image
image

- ~added a caption above the header text~  removed from this PR
- added a meta-data text option
- ~wrapped the expansion-panel button with H3 - considering wether it's too generic and we need to set is as default with an option to change the headline number according to the component position in the page hierarchy.~ **moved to [1228](https://github.com/Vonage/vivid/issues/1228)**
- fixed dense ui
- limited long header to wrap only into 2 lines:
image

### CODE REVIEW

File: components/expansion-panel/test/expansion-panel.test.js [hunk 1]
```diff
@@ -80,6 +80,35 @@ describe('expansion panel', () => {
...
+		expect(meta.textContent.trim()).to.equal(metaText);
+	});
+
+	it('expansion-panel-header should have meta class when meta is set', async () => {
+		const metaText = 'meta-data';
+		const [actualElement] = addElement(
+			textToDomToParent(`<${COMPONENT_NAME} metaData="${metaText}"></${COMPONENT_NAME}>`)
+		);
+		await waitNextTask();
+		expect(actualElement.shadowRoot.querySelector('.meta'));
```
Comment (YonatanKra) [inline components/expansion-panel/test/expansion-panel.test.js:109]: Expect what?

File: components/expansion-panel/test/expansion-panel.test.js [hunk 2]
```diff
@@ -80,6 +80,35 @@ describe('expansion panel', () => {
...
+		const metaText = 'meta-data';
+		const [actualElement] = addElement(
+			textToDomToParent(`<${COMPONENT_NAME} metaData="${metaText}"></${COMPONENT_NAME}>`)
+		);
+		await waitNextTask();
+		const meta = actualElement.shadowRoot.querySelector('.expansion-panel-header');
+		expect(meta.textContent.trim()).to.equal(metaText);
+	});
+
+	it('expansion-panel-header should have meta class when meta is set', async () => {
```
Comment (YonatanKra) [inline components/expansion-panel/test/expansion-panel.test.js:103]: Always start with `should`. Remove `expansion-panel-header`.
Maybe you meant this:
```suggestion
	it('should set meta class on expansion-panel-header when meta is set', async () => {
```
?

File: components/expansion-panel/test/expansion-panel.test.js [hunk 3]
```diff
@@ -80,6 +80,35 @@ describe('expansion panel', () => {
...
+		const captionText = 'Item caption';
+		const [actualElement] = addElement(
+			textToDomToParent(`<${COMPONENT_NAME} caption="${captionText}"></${COMPONENT_NAME}>`)
+		);
+		await waitNextTask();
+		const caption = actualElement.shadowRoot.querySelector('.expansion-panel-header');
+		expect(caption.textContent.trim()).to.equal(captionText);
+	});
+
+	it('should have meat text when meta is set', async () => {
```
Comment (YonatanKra) [inline components/expansion-panel/test/expansion-panel.test.js:93]:
```suggestion
	it('should have meta text when meta is set', async () => {
```

File: components/expansion-panel/test/expansion-panel.test.js [hunk 4]
```diff
@@ -80,6 +80,35 @@ describe('expansion panel', () => {
...
+		const metaText = 'meta-data';
+		const [actualElement] = addElement(
+			textToDomToParent(`<${COMPONENT_NAME} metaData="${metaText}"></${COMPONENT_NAME}>`)
+		);
+		await waitNextTask();
+		const meta = actualElement.shadowRoot.querySelector('.expansion-panel-header');
+		expect(meta.textContent.trim()).to.equal(metaText);
```
Comment (YonatanKra) [inline components/expansion-panel/test/expansion-panel.test.js:100]: What happens if we have both `meta` and `text`?

File: components/expansion-panel/test/expansion-panel.test.js (hunk 1 above)
Comment (rachelbt) [reply components/expansion-panel/test/expansion-panel.test.js:109]: yeh - this one was tricky for me. the `.expansion-panel-header` need to have also the class `.meta`

File: components/expansion-panel/test/expansion-panel.test.js (hunk 4 above)
Comment (yinonov) [reply components/expansion-panel/test/expansion-panel.test.js:100]: See #1126

Comment (yinonov) [reply components/expansion-panel/test/expansion-panel.test.js:100]: Text should actually be named as label

Comment (rachelbt) [reply components/expansion-panel/test/expansion-panel.test.js:100]: why label?

Comment (yinonov) [reply components/expansion-panel/test/expansion-panel.test.js:100]: It's the button's label

Comment (yinonov) [reply components/expansion-panel/test/expansion-panel.test.js:100]: Can easily support the 2 while deprecating text already

Comment (rachelbt) [reply components/expansion-panel/test/expansion-panel.test.js:100]: but the caption is not the button label the `header` is...

Comment (rachelbt) [reply components/expansion-panel/test/expansion-panel.test.js:104]: I think we are talking about something that is not existing. There is no "text" property.
theses are the "text areas" that have in the expansion panel:
header, caption, meta-data
image
I'm not sure that caption needs to be replaced with label.

File: components/expansion-panel/src/vwc-expansion-panel.ts [hunk 5]
```diff
@@ -52,7 +54,13 @@ export class VWCExpansionPanel extends VWCExpansionPanelBase {
 	@property({type: String, reflect: true})
 		indicatorIconSet: IndicatorIconSets[number] = 'chevron';
 
-	@property({type: Boolean, reflect: true})
+	@property({ type: String, reflect: true })
+		caption = '';
+
+	@property({ type: String, reflect: true })
+		metaData = '';
```
Comment (yinonov) [inline components/expansion-panel/src/vwc-expansion-panel.ts:61]:
```suggestion
		meta = '';
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/test/expansion-panel.test.js (hunk 4 above)
Comment (rachelbt) [reply components/expansion-panel/test/expansion-panel.test.js:108]: removed the caption from this branch

File: components/expansion-panel/src/vwc-expansion-panel.ts (hunk 5 above)
Comment (rachelbt) [reply components/expansion-panel/src/vwc-expansion-panel.ts]: changed it.

File: __snapshots__/expansion panel.md [hunk 6]
```diff
@@ -7,14 +7,17 @@
...
+    id="expansion-panel"
   >
     <mwc-ripple>
     </mwc-ripple>
     <span class="leading-icon">
       <slot name="icon">
       </slot>
     </span>
-    click me
+    <span class="header-text">
```
Comment (yinonov) [inline __snapshots__/expansion panel.md:18]: wasn't this corrected to heading?

File: components/expansion-panel/src/vwc-expansion-panel.scss [hunk 7]
```diff
@@ -50,6 +50,31 @@
...
+		display: -webkit-box;
+		overflow: hidden;
+		flex-grow: 1;
+		-webkit-box-orient: vertical;
+		-webkit-line-clamp: 2;
+		/* stylelint-enable property-no-vendor-prefix */
+		/* stylelint-enable value-no-vendor-prefix */
+		margin-inline-end: auto;
+		text-align: left;
+	}
```
Comment (yinonov) [inline components/expansion-panel/src/vwc-expansion-panel.scss:65]: what's this for? are we allowing more than a single line for heading?

File: __snapshots__/expansion panel.md (hunk 6 above)
Comment (rachelbt) [reply __snapshots__/expansion panel.md]: leftovers... changed the class name

File: components/expansion-panel/src/vwc-expansion-panel.scss (hunk 7 above)
Comment (rachelbt) [reply components/expansion-panel/src/vwc-expansion-panel.scss]: Yes. up to 2 lines as. Joella said

File: __snapshots__/expansion panel.md [hunk 8]
```diff
@@ -7,14 +7,17 @@
...
+    id="expansion-panel"
   >
     <mwc-ripple>
     </mwc-ripple>
     <span class="leading-icon">
       <slot name="icon">
       </slot>
     </span>
-    click me
+    <span class="heading-text">
```
Comment (yinonov) [inline __snapshots__/expansion panel.md:18]: it's no longer heading text right? it's the button's label.
it's implementation details and not an issue but maybe deserves a better name for maintainers legibility

**Review by yinonov — CHANGES_REQUESTED** [review]

File: components/expansion-panel/src/vwc-expansion-panel.scss (hunk 7 above)
Comment (yinonov) [reply components/expansion-panel/src/vwc-expansion-panel.scss]: can we extract this to another issue/PR?

File: __snapshots__/expansion panel.md (hunk 8 above)
Comment (rachelbt) [reply __snapshots__/expansion panel.md:18]: It is the button label - but in the expansion-panel is is the headline of the expansion. 
Its only a class name for our use.

Comment (yinonov) [reply __snapshots__/expansion panel.md:18]: I know, missed it in the previous PR. NM

File: components/expansion-panel/src/vwc-expansion-panel.scss [hunk 9]
```diff
@@ -50,6 +50,31 @@
...
```
Comment (yinonov) [inline components/expansion-panel/src/vwc-expansion-panel.scss:65]: can you please open a ticket to discuss the number of lines support?

**Review by yinonov — APPROVED** [review]

File: components/expansion-panel/src/vwc-expansion-panel.scss (hunk 9 above)
Comment (rachelbt) [reply components/expansion-panel/src/vwc-expansion-panel.scss]: removed the line-trim and opened an issue: https://github.com/Vonage/vivid/issues/1246

**Review by yinonov — APPROVED** [review]

**Pull request merged**